src/main/java/com/svydovets/bibirnate/cache/Cache.java

```diff
@@ -0,0 +1,27 @@
+package com.svydovets.bibirnate.cache;
+
+import com.svydovets.bibirnate.util.EntityKey;
+
+import java.util.Optional;
+
+/**
+ * The cache provides opportunities to save an entity and in case when application need it id kan be provided by a key,
+ * as a key the Bibernate uses for this purpose {@link EntityKey}. Also, {@link Cache} provides some methods for
+ * working with this cache.
+ */
+public interface Cache {
+
+    <T> void put(EntityKey<T> entityKey, T entity);
```

Comment:
Please move javadocs to the interface methods and @inheritDoc in the implementation.